pkg/codegen/schema/mock_pulumi_schema.go

```diff
@@ -38,8 +34,7 @@ func newPulumiPackage() *Package {
 	}
 
 	pkg, err := ImportSpec(spec, nil)
-	contract.AssertNoError(err)
-	return pkg
+	return pkg, err
 }
 
-var DefaultPulumiPackage = newPulumiPackage()
+var DefaultPulumiPackage, DefaultPulumiPackageError = newPulumiPackage()
```

Comment:
I can see why we need to expose the error information (thanks for the reference to the lsp example), but I think this is a bit fragile. It's easy to forget to check the error here. It would be preferable if we could make this a function.

If it's possible to change this API, we can hide the variable and introduce a top-level function:

```go
func DefaultPulumiPackage() (*Package, error)
```

Implementation would internally use sync.Once to maintain the singleton:

```go
var defaultPulumiPackage struct {
  pkg *Package
  err error
  once sync.Once
}

func DefaultPulumiPackage() (*Package, error) {
  defaultPulumiPackage.once.Do(func() {
    defaultPulumiPackage.pkg, defaultPulumiPackage.err = newPulumiPackage()
  })
  return defaultPulumiPackage.pkg, defaultPulumiPackage.err
}
```

Reply:
newPulumiPackage _should not fail_, it's an engine bug if it errors.